projects/blenderbot2/agents/blenderbot2.py

```diff
@@ -800,7 +800,10 @@ def _set_batch_skip_search(self, valid_exs: List[Message], batch: Batch) -> Batc
 
     def eval_step(self, batch):
         output = super().eval_step(batch)
-        if output is None or not hasattr(self.model, 'retriever'):
+        model = self.model
+        if isinstance(self.model, torch.nn.parallel.DistributedDataParallel):
```

Comment:
actually, this is what `self.model_api` is meant to accomplish, if you can just switch to that